CheckerUI_xforms/CheckerUI/CheckerUI/CheckerUI/ViewModels/BaseLineViewModel.cs

```diff
@@ -78,88 +52,158 @@ private void fillLayout()
             ;
             for (int i = m_counterID; i < m_counterID + 3; i++)
             {
-                if (m_GridToMake.Children.Count < 5)
-                {
-                    int status = -1;
-                    buttonToMake = new Button();
-                    buttonToMake.Padding = new Thickness(20, 20, 20, 20);
-                    buttonToMake.Text = i.ToString();
-                    buttonToMake.Clicked += Button_Clicked;
-                    buttonToMake.CornerRadius = 10;
-                    buttonToMake.BackgroundColor = Color.Gold;
-                    OrderIDNotifier m = new OrderIDNotifier(i, status);
-                    m_GridToMake.Children.AddVertical(buttonToMake);
-                    order = new OrderItemView(nmList[nameIdx++], buttonToMake, m);
-
-                    m.PropertyChanged += M_PropertyChanged;
-                    m_Orders.Add(i, order);
-                }
-                else
-                {
-                    // add to waiting list 
-                }
+
+                int status = 0;
+                buttonToMake = new Button();
+                buttonToMake.Padding = new Thickness(20, 20, 20, 20);
+                buttonToMake.Text = i.ToString();
+                buttonToMake.Clicked += Button_Clicked;
+                buttonToMake.CornerRadius = 10;
+                buttonToMake.BackgroundColor = Color.Gold;
+                buttonToMake.IsVisible = true;
+                OrderIDNotifier m = new OrderIDNotifier(i, status);
+                OrderButtonModel model = new OrderButtonModel(i, buttonToMake);
+                m_ButtonsToMake.Add(model);
+                order = new OrderItemView(nmList[nameIdx++], model, m);
+                m.PropertyChanged += OrderIDNotifierPropertyChanged;
+                m_Orders.Add(i, order);
+
             }
-          
             m_counterID += 3;
+            buttonToMake = new Button();
+            buttonToMake.Padding = new Thickness(20, 20, 20, 20);
+            buttonToMake.Text = m_LockedCounterID.ToString();
+            buttonToMake.Clicked += Button_Clicked;
+            buttonToMake.CornerRadius = 10;
+            buttonToMake.BackgroundColor = Color.Gold;
+            buttonToMake.IsVisible = true;
```

Comment:
Also, since these are very particular examples of these buttons, can they not be defined as internal classes and initialized in that way?
such as:
``` 
internal class someButton : Button{
    public someButton(string i_Text, Command i_Command){
        setDesign();
        Text = i_Text;
        Clicked = i_Command;
    }
    private void setDesign(){
        Padding = new Thickness(20, 20, 20, 20);
        CornerRadius = 20;
        ...
    }
}

Reply:
Yes, in fact it is already like that because there we have model for ordering with a button, it's just an unnecessary duplication that can already be arranged, handle it